testing-common/src/test/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollectorTest.java

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.muzzle.collector;
+
+import static io.opentelemetry.javaagent.extension.muzzle.Flag.MinimumVisibilityFlag.PACKAGE_OR_HIGHER;
+import static io.opentelemetry.javaagent.extension.muzzle.Flag.MinimumVisibilityFlag.PROTECTED_OR_HIGHER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import external.instrumentation.ExternalHelper;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.InstrumentationContextTestClasses;
+import io.opentelemetry.instrumentation.OtherTestHelperClasses;
+import io.opentelemetry.instrumentation.TestHelperClasses;
+import io.opentelemetry.javaagent.extension.muzzle.ClassRef;
+import io.opentelemetry.javaagent.extension.muzzle.FieldRef;
+import io.opentelemetry.javaagent.extension.muzzle.Flag;
+import io.opentelemetry.javaagent.extension.muzzle.Flag.ManifestationFlag;
+import io.opentelemetry.javaagent.extension.muzzle.Flag.OwnershipFlag;
+import io.opentelemetry.javaagent.extension.muzzle.Flag.VisibilityFlag;
+import io.opentelemetry.javaagent.extension.muzzle.MethodRef;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import muzzle.DeclaredFieldTestClass;
+import muzzle.TestClasses;
+import muzzle.TestClasses.HelperAdvice;
+import muzzle.TestClasses.LdcAdvice;
+import muzzle.TestClasses.MethodBodyAdvice;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import spock.lang.Specification;
+
+class ReferenceCollectorTest extends Specification {
+
+  @Test
+  public void methodBodyCreatesReferences() {
+    ReferenceCollector collector = new ReferenceCollector((String s) -> false);
+
+    collector.collectReferencesFromAdvice(MethodBodyAdvice.class.getName());
+    collector.prune();
+    Map<String, ClassRef> references = collector.getReferences();
+
+    assertEquals(
```

Comment:
We use assertj (which will also make things like container assertions much simpler)